tests/test_core_network.py

```diff
@@ -3008,3 +3008,10 @@ def test_validation_report_for_node_elevation(network4):
                       'values': {'extremely_high_values_dict': {}, 'extremely_low_values_dict': {'101982': -51}}}
 
     assert_semantically_equal(report, correct_report)
+
+
+def test_validation_report_for_network_without_node_elevation(network3):
+    n = network3
+    with pytest.raises(Exception) as error_info:
```

Comment:
I think you want to be very specific about the exact type of exception you expect here, otherwise the test could be failing for reasons other than the one you expect and you would not know it. You want the test to _pass_ when the expected `MissingElevationException` is thrown, but you want it to _fail_ if any other type of exception is thrown, so this should become `with pytest.raises(MissingElevationException) as error_info`.

Also, why did you place the exception class definition inside a function? I am not sure what the implications of that are - is it visible outside of that function? It seems like an odd place to put it.